samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -217,8 +225,9 @@ private void generateNewJobModel() {
     log.info("pid=" + processorId + "published new JobModel ver=" + nextJMVersion + ";jm=" + jobModel);
 
     // start the barrier for the job model update
-    ZkBarrierForVersionUpgrade barrier = (ZkBarrierForVersionUpgrade) coordinationUtils.getBarrier(JOB_MODEL_VERSION_BARRIER);
-    barrier.start(nextJMVersion, currentProcessors);
+    ZkBarrierForVersionUpgrade barrier = (ZkBarrierForVersionUpgrade) coordinationUtils.getBarrier(
```

Comment:
QQ: Why do we need an explicit casting here ? Is the interface for barrier not the same as that returned by  CoordinationUtils?